newIDE/app/src/ResourcesList/FileToCloudProjectResourceUploader.js

```diff
@@ -0,0 +1,206 @@
+  audio: 'audio/*',
+  image: 'image/*',
+  font: 'font/*',
+  video: 'video/*',
+  json: 'application/json',
+  bitmapFont: '.fnt,.xml',
+};
+
+export const FileToCloudProjectResourceUploader = ({
+  options,
+  fileMetadata,
+  getStorageProvider,
+  onChooseResources,
+  createNewResource,
+}: FileToCloudProjectResourceUploaderProps) => {
+  const inputRef = React.useRef<?HTMLInputElement>(null);
+  const authenticatedUser = React.useContext(AuthenticatedUserContext);
+  const [isUploading, setIsUploading] = React.useState(false);
+  const [selectedFiles, setSelectedFiles] = React.useState<File[]>([]);
+  const hasSelectedFiles = selectedFiles.length > 0;
+  const storageProvider = React.useMemo(getStorageProvider, [
+    getStorageProvider,
+  ]);
+  const cloudProjectId = fileMetadata.fileIdentifier;
+  const [uploadProgress, setUploadProgress] = React.useState(0);
+  const onUpload = React.useCallback(
+    async () => {
+      const input = inputRef.current;
+      if (!input) return;
+
+      try {
+        setIsUploading(true);
+        setUploadProgress(0);
+        const results: UploadedProjectResourceFiles = await uploadProjectResourceFiles(
+          authenticatedUser,
+          cloudProjectId,
+          selectedFiles,
+          (current: number, total: number) => {
+            setUploadProgress((current / total) * 100);
+          }
+        );
+        const erroredResults = results.filter(({ error }) => !!error);
+        const okResults = results.filter(({ url }) => !!url);
+        if (erroredResults.length) {
+          throw erroredResults[0];
+        } else if (okResults.length) {
+          onChooseResources(
+            okResults.map(({ url, resourceFile }) => {
+              const newResource = createNewResource();
+              newResource.setFile(url || '');
+              newResource.setName(resourceFile.name);
+              newResource.setOrigin('cloud-project-resource', url || '');
+
+              return newResource;
+            })
+          );
+        }
+      } catch (error) {
+        showErrorBox({
+          message:
+            'There was an error while uploading some resources. Verify your internet connection or try again later.',
+          rawError: error,
+          errorId: 'upload-cloud-project-resource-error',
+          doNotReport: true,
+        });
+      } finally {
+        setIsUploading(false);
+      }
+    },
+    [
+      selectedFiles,
+      authenticatedUser,
+      onChooseResources,
+      createNewResource,
+      cloudProjectId,
+    ]
+  );
+
+  const invalidFiles = selectedFiles
+    .map(file => {
+      if (file.size > PROJECT_RESOURCE_MAX_SIZE_IN_BYTES) {
+        return {
+          filename: file.name,
+          error: 'too-large',
+        };
+      }
+      return null;
+    })
+    .filter(Boolean);
+
+  const canUploadWithThisStorageProvider =
+    storageProvider.internalName === 'Cloud';
+  const isConnected = !!authenticatedUser.profile;
```

Comment:
`authenticatedUser.authenticated` doesn't work for this?

Reply:
It does work, I just forgot about this! Thanks.